tests/unit/internal-session-test.js

```diff
@@ -776,6 +776,28 @@ describe('InternalSession', () => {
               done();
             });
           });
+
+          it('it does not trigger the "sessionInvalidationFailed" event', function(done) {
+            let triggered = false;
+            sinon.stub(session, '_lookupAuthenticator').returns(authenticator);
+            session.one('sessionInvalidationFailed', () => (triggered = true));
+
+            session.invalidate().then(() => {
+              expect(triggered).to.be.false;
+              done();
+            });
+          });
```

Comment:
This can be shortened as below by returning the promise instead of using `done` and removing the stub.